### PR TITLE
fix: trigger `onDisconnect` when removing a wallet

### PR DESCRIPTION
**Pull request by neuodev**
<details><summary>Preview</summary>
<p>

https://github.com/Emurgo/yoroi-frontend/assets/72753578/6ceafaef-bcb7-4aa2-a524-14ee64a86c0c

</p>
</details>

### CODE REVIEW

**Comment by neuodev** [comment]
[autoscirpt] Jira Ticket: **[OnDisconnect isn't called in case of deleting the connected wallet](https://emurgo.atlassian.net/browse/YOEXT-545)**

[](__JIRA_COMMENT_FLAG)

**Review by Nebyt — APPROVED** [review]

**Review by zuzunker — APPROVED** [review]
Nice! Thank you, @neuodev !

**Review by zuzunker — COMMENTED** [review]
/check

File: packages/yoroi-extension/app/stores/toplevel/WalletSettingsStore.js [hunk 1]
```diff
@@ -285,9 +285,22 @@ export default class WalletSettingsStore extends Store<StoresMap, ActionsMap> {
...
+    const connectedDapps = connectorWhitelist.filter(
+      dapp => dapp.publicDeriverId === request.publicDeriver.publicDeriverId
+    );
+
+    for (const dapp of connectedDapps) {
+      await this.actions.connector.removeWalletFromWhitelist.trigger({
+        protocol: dapp.protocol,
+        url: dapp.url,
+      });
+    };
```
Comment (zuzunker) [inline packages/yoroi-extension/app/stores/toplevel/WalletSettingsStore.js]:
```suggestion
    }
```

**Review by zuzunker — COMMENTED** [review]
/check

**Pull request merged**